test/build-images.sh

```diff
@@ -40,20 +32,24 @@ then
   echo "docker images for api-server, frontend, scheduledworkflow and \
     persistenceagent are already built in ${GCR_IMAGE_BASE_DIR}."
 else
-  echo "submitting argo workflow to build docker images for commit ${PULL_PULL_SHA}..."
-  # Build Images
-  ARGO_WORKFLOW=`argo submit ${DIR}/build_image.yaml \
-  -p image-build-context-gcs-uri="$remote_code_archive_uri" \
-  ${IMAGE_BUILDER_ARG} \
-  -p api-image="${GCR_IMAGE_BASE_DIR}/api-server" \
-  -p frontend-image="${GCR_IMAGE_BASE_DIR}/frontend" \
-  -p scheduledworkflow-image="${GCR_IMAGE_BASE_DIR}/scheduledworkflow" \
-  -p persistenceagent-image="${GCR_IMAGE_BASE_DIR}/persistenceagent" \
-  -n ${NAMESPACE} \
-  --serviceaccount test-runner \
-  -o name
-  `
-  echo "build docker images workflow submitted successfully"
-  source "${DIR}/check-argo-status.sh"
-  echo "build docker images workflow completed"
+  echo "submitting cloud build to build docker images for commit ${PULL_PULL_SHA}..."
+  IMAGES_BUILDING=true
+  CLOUD_BUILD_COMMON_ARGS=(. --async --format='value(id)' --substitutions=_GCR_BASE=${GCR_IMAGE_BASE_DIR})
+  # Use faster machine because this is CPU intensive
+  BUILD_ID_API_SERVER=$(gcloud builds submit ${CLOUD_BUILD_COMMON_ARGS[@]} \
```

Comment:
is it possible to combine these steps to a single file?
cloud build seems support parallel execution
waitFor: ['-']

Reply:
https://github.com/GoogleCloudPlatform/cloudbuild-integration-testing/pull/41/files

Reply:
Yes, I also tried that.
There are some tradeoffs using waitFor: ['-']:
* [major] slower because cloud build will run all the image builds in one machine, they compete for resources. Time taken is ~9min -> ~12min. Consider that we may need to build more images later. I think current approach is better. (or I can batch 3 image builds and only let api-server use a single job)
* [minor] log will interleave for 4 builds, a little harder to read

Reply:
I prefer
* batch the 3 fast image builds
* let api-server image build still be standalone

Reply:
Made corresponding changes.

Reply:
SG. BTW with this change https://github.com/kubeflow/pipelines/pull/1904 I'm not sure if building api-server images still consumes lots of resources. the change removed most of the dependencies such as tensorflow.

Reply:
Thanks for the heads up. I will take a look and change if building api-server is faster.